go/teams/teams.go

```diff
@@ -465,29 +476,33 @@ func (t *Team) sigChangeItem(ctx context.Context, section SCTeamSection, linkTyp
 		return libkb.SigMultiItem{}, err
 	}
 	sig, err := ChangeSig(me, latestLinkID1, t.NextSeqno(), deviceSigningKey, section, linkType, merkleRoot)
-	if err != nil {
-		return libkb.SigMultiItem{}, err
-	}
 
-	signingKey, err := t.keyManager.SigningKey()
-	if err != nil {
-		return libkb.SigMultiItem{}, err
-	}
-	encryptionKey, err := t.keyManager.EncryptionKey()
 	if err != nil {
 		return libkb.SigMultiItem{}, err
 	}
 
-	if section.PerTeamKey != nil {
-		// need a reverse sig
-
-		// set a nil value (not empty) for reverse_sig (fails without this)
-		sig.SetValueAtPath("body.team.per_team_key.reverse_sig", jsonw.NewNil())
-		reverseSig, _, _, err := libkb.SignJSON(sig, signingKey)
+	var signingKey libkb.NaclSigningKeyPair
+	var encryptionKey libkb.NaclDHKeyPair
+	if linkType != libkb.LinkTypeLeave {
```

Comment:
My intent here and later in this method was to be explicit about the cases where we use `PerTeamKey`s, my fear was that due to some bug in the future `encryptionKey`/`signingKey` could be `nil` for a link type that does need it and it wouldn't get caught until it goes to the server. Failing faster might be a good thing.

I was thinking a cleaner solution could be abstracted out into a function that takes in a linktype and returns whether it uses `PerTeamKey`s.